### PR TITLE
Adopting graveyard convention for DAGMC models.

### PR DESCRIPTION
**Pull request by pshriwise**
DAGMC uses the "graveyard" to identify a CAD volume containing all other volumes of interest as a representation of the boundary between the model and the rest of the virtual world. 

Graveyard volumes in DAGMC are identified as having a name `mat:Graveyard`.

This PR adopts this naming convention used in other DAGMC interfaces and sets the boundary conditions of the graveyard surfaces accordingly (vacuum boundary) to remove particles from the simulation when they enter this region. Using this convention in OpenMC will allow DAGMC models with the UWUW workflow to run in OpenMC as they would in any other code supporting DAGMC models without modification.

### CODE REVIEW

File: src/dagmc.cpp [hunk 1]
```diff
@@ -231,6 +236,11 @@ void load_dagmc_geometry()
     model::overlap_check_count.resize(model::cells.size(), 0);
   }
 
+  if (!graveyard) {
+    std::cout << "WARNING: No graveyard volume found in the DagMC model.\n";
+    std::cout << "This may result in lost particles and rapid simulation failure.\n";
```
Comment (paulromano) [inline src/dagmc.cpp]:
```C++
warning("No graveyard volume...");
```

File: src/dagmc.cpp [hunk 2]
```diff
@@ -231,6 +236,11 @@ void load_dagmc_geometry()
     model::overlap_check_count.resize(model::cells.size(), 0);
   }
 
+  if (!graveyard) {
+    warning("No graveyard volume found in the DagMC model.");
+    warning("This may result in lost particles and rapid simulation failure.");
```
Comment (paulromano) [inline src/dagmc.cpp]: This will result in two separate warnings. Just have the two strings implicitly joined within one call to `warning()`

**Comment by makeclean** [comment]
I would maybe consider switching to DMD for the metadata, rather than parse props directly. Thats unrelated to this PR, just a general request.

**Comment by makeclean** [comment]
But otherwise it looks great to me

**Comment by pshriwise** [comment]
@makeclean thanks for taking a look! 

The DMD is a little tricky for this case. It works great for the UWUW workflow, but requires a density value in the material assignment for legacy workflows. Here, using the legacy workflow, materials will be assigned a density in the `materials.xml` file rather than as part of the material assignment. This makes it difficult to use the DMD for a legacy material assignments - if there are densities on the cells and the materials, which should be used?

I *think* that cell temperatures currently override material temperatures if set, so we could follow that protocol here but we would probably want that to be true for native geometry as well. This would lead to shuffling in the order of xs data loading to resolve density assignments etc... In short, it gets complicated unless we're ok with simply ignoring density assignments if present in the DAGMC file or don't require them to exist - which I don't think the DMD allows.

Those are my thoughts. As @makeclean suggested, we can perhaps resolve this in a later PR.

**Comment by pshriwise** [comment]
Also, I caught an oversight along the way for non-void implicit complements (ICs)- this feature would have only worked with UWUW. Now it should work for legacy formats of IC material assignment as well.

**Pull request merged**